x-pack/libbeat/management/managerV2.go

```diff
@@ -260,7 +260,7 @@ func (cm *BeatV2Manager) unitListen() {
 				cm.logger.Debugf("Got unit modified: %s, type: %s expected state: %s", change.Unit.ID(), change.Unit.Type(), state.String())
 				// I'm assuming that a state STOPPED just tells us to shut down the entire beat,
 				// as such we don't really care about updating via a particular unit
-				if state == client.UnitStateStopped {
+				if state == client.UnitStateStopped || state == client.UnitStateStopping {
```

Comment:
From `Expected()` you will either receive `HEALTHY` or `STOPPED` that is the only 2 expected states that the Elastic Agent will tell you for a Unit. Of course the actual state of the unit can be of any type.

When you get `client.UnitstateStopped` it is telling that only that specific unit should be stopped. That does not mean the whole beat should be stopped. Example is when Elastic Agent is running 2 filestream inputs, and then only 1 is removed. The remove unit would be changed to `client.UnitStateStopped` where the other unit would remain as `client.UnitStateHealthy`. Elastic Agent will still report that removed unit as existing until you report `client.UnitStateStopped` through `unit.UpdateState`. Once that is done the an update from the client to fully remove the unit will occur.

Reply:
> From Expected() you will either receive HEALTHY or STOPPED that is the only 2 expected states that the Elastic Agent will tell you for a Unit. 

Ah, thanks, I was a bit confused by the logic there.

Reply:
So `UnitStateStopping` is only used for clients to update state to the server?

Reply:
`UnitStateStopping` is an information only state. It tells the Elastic Agent that work is being done to stop the unit. If the unit can be stopped immediately you can just report `UnitStateStopped` instantly (`UnitStateStopping` is not required to be reported).

Once `UnitStateStopped` is reported from `Expected()` it is required that the unit report as `UnitStateStopped` once it has completely stopped.